dvc/cli.py

```diff
@@ -205,6 +375,10 @@ def get_main_parser():
 
     fix_subparsers(subparsers)
 
+    subparsers.add_parser = types.MethodType(
+        add_parser_exit_on_error, subparsers
+    )
```

Comment:
Decided to override the behavior of `add_parser` here in one place for all subparsers.

The logic in `add_parser_exit_on_error()` is exactly the same as in the **argparse** package with the exception of the `sys.version_info >= (3, 9, 0)` check in the beginning that sets `exit_on_error=False`.

I believe this approach is cleaner than manually updating each of the `subparsers.add_parser()` calls. As for the new unit tests, the `@pytest.mark.skipif` decorator means these tests won't run when running tests in environments less than Python 3.9.

Reply:
Also, this automatically handles any new DVC commands that are added. For example, the new **stage** command is displayed as a suggestion:

![Screen Shot 2021-01-20 at 3 58 40 PM](https://user-images.githubusercontent.com/7004949/105255436-63776f80-5b38-11eb-9a7c-8fabcc7e112a.png)